src/cmap/auth/mongodb_aws.ts

```diff
@@ -138,8 +138,8 @@ export class MongoDBAWS extends AuthProvider {
     );
 
     const payload: AWSSaslContinuePayload = {
-      a: options.headers.Authorization,
-      d: options.headers['X-Amz-Date']
+      a: `${options.headers?.Authorization ?? ''}`,
+      d: `${options.headers?.['X-Amz-Date'] ?? ''}`
```

Comment:
after using the typings from `@types/aws4`, `options.headers` becomes optional. converting those values to hard-coded string type.